packages/graph/hash_graph/lib/graph/src/api/rest/link_type.rs

```diff
@@ -0,0 +1,172 @@
+        .create_link_type(body.schema, body.account_id)
+        .await
+        .map_err(|report| {
+            if report.contains::<BaseUriAlreadyExists>() {
+                return StatusCode::CONFLICT;
+            }
+
+            // Insertion/upddate errors are considered internal server errors.
+            StatusCode::INTERNAL_SERVER_ERROR
+        })
+        .map(Json)
+}
+
+#[utoipa::path(
+    get,
+    path = "/link-type/{versionId}",
+    tag = "LinkType",
+    responses(
+        (status = 200, content_type = "application/json", description = "Link type found", body = QualifiedLinkType),
+        (status = 422, content_type = "text/plain", description = "Provided version_id is invalid"),
+
+        (status = 404, description = "Link type was not found"),
+        (status = 500, description = "Datastore error occurred"),
+    ),
+    params(
+        ("versionId" = Uuid, Path, description = "The version ID of link type"),
+    )
+)]
+async fn get_link_type<D: Datastore>(
+    version_id: Path<Uuid>,
+    datastore: Extension<D>,
+) -> Result<Json<Qualified<LinkType>>, impl IntoResponse> {
+    let Path(version_id) = version_id;
+    let Extension(datastore) = datastore;
+
+    datastore
+        .get_link_type(VersionId::new(version_id))
+        .await
+        .map_err(|report| {
+            if report.contains::<QueryError>() {
+                return StatusCode::NOT_FOUND;
+            }
+
+            // Datastore errors such as connection failure are considered internal server errors.
+            StatusCode::INTERNAL_SERVER_ERROR
+        })
+        .map(Json)
+}
+
+// async fn get_link_type_many() -> Result<String, StatusCode> {
+//     unimplemented!()
+// }
+
+#[derive(Component, Serialize, Deserialize)]
+struct UpdateLinkTypeRequest {
+    #[component(value_type = Any)]
+    schema: LinkType,
+    account_id: AccountId,
+}
+
+#[utoipa::path(
+    put,
+    path = "/link-type",
+    tag = "LinkType",
+    responses(
+        (status = 200, content_type = "application/json", description = "Link type updated successfully", body = QualifiedLinkType),
+        (status = 422, content_type = "text/plain", description = "Provided request body is invalid"),
+
+        (status = 404, description = "Base link type ID was not found"),
+        (status = 500, description = "Datastore error occurred"),
+    ),
+    request_body = UpdateLinkTypeRequest,
+)]
+async fn update_link_type<D: Datastore>(
+    body: Json<UpdateLinkTypeRequest>,
+    datastore: Extension<D>,
+) -> Result<Json<Qualified<LinkType>>, StatusCode> {
+    let Json(body) = body;
+    let Extension(datastore) = datastore;
+
+    datastore
+        .clone()
+        .update_link_type(body.schema, body.account_id)
+        .await
+        .map_err(|report| {
+            if report.contains::<BaseUriDoesNotExist>() {
+                return StatusCode::NOT_FOUND;
+            }
+
+            // Insertion/upddate errors are considered internal server errors.
```

Comment:
```suggestion
            // Insertion/update errors are considered internal server errors.
```
I noticed this typo I made, I suppose it's made at least 5 other places :(